File: src/main/java/com/example/chatService/model/ChatMessage.java
package com.example.chatService.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;

@Data
@Entity
@Table(name = "chat_message")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String roomId;
    private String sender;
    private String recipient;
    private String content;
    @Enumerated(EnumType.STRING)
    private MessageType type;

    private OffsetDateTime timestamp;

    @PrePersist
    public void prePersist() {
        timestamp = OffsetDateTime.now();
    }

    public ChatMessage(String content, String sender, MessageType type) {
        this.content = content;
        this.sender = sender;
        this.type = type;
    }

    public enum MessageType {
        CHAT, JOIN, LEAVE, DM
    }
}

File: src/main/java/com/example/chatService/model/ChatMessage.kt
package com.example.chatService.model

import jakarta.persistence.*
import java.time.OffsetDateTime

@Entity
@Table(name = "chat_message")
class ChatMessage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var roomId: String? = null,
    var sender: String? = null,
    var recipient: String? = null,
    var content: String? = null,
    @Enumerated(EnumType.STRING)
    var type: MessageType? = null,
    var timestamp: OffsetDateTime? = null
) {
    @PrePersist
    fun prePersist() {
        timestamp = OffsetDateTime.now()
    }

    constructor(content: String?, sender: String?, type: MessageType?) : this(
        content = content,
        sender = sender,
        type = type
    )

    enum class MessageType {
        CHAT, JOIN, LEAVE, DM
    }
}
